packages/web-components/fast-components/src/radio/radio.template.ts

```diff
@@ -0,0 +1,25 @@
+import { html, when } from "@microsoft/fast-element";
+import { Radio } from "./radio";
+
+export const RadioTemplate = html<Radio>`
+    <template
+        role="radio"
+        aria-checked="${x => x.checked}"
+        aria-required="${x => x.required}"
+        aria-disabled="${x => x.disabled}"
+        aria-readonly="${x => x.readOnly}"
+        tabindex="${x => (x.disabled ? null : 0)}"
+        @keypress="${(x, c) => x.keypressHandler(c.event as KeyboardEvent)}"
+        @click="${(x, c) => x.clickHandler(c.event as MouseEvent)}"
+    >
+        <div part="control" class="control">
+            <slot name="checked-indicator">
+                <div part="checked-indicator" class="checked-indicator"></div>
+            </slot>
+        </div>
+        ${when(
+            x => x.childNodes.length,
+            html` <label part="label" class="label"><slot></slot></label> `
+        )}
```

Comment:
Have we tested and validated and tested that this behaves as expected? Just curious based off my exchange here in the W3C repo: https://github.com/w3c/aria-practices/issues/1286#issuecomment-566372426

Reply:
Chris looking at that bug it doesn't seem like the same scenario as the template code used here and in checkbox, that refers to a label outside the radio tree/branch and using the for attribute. in this case it's just content of the radio. At least the voice over on Mac is reading it fine. I don't have JAWS on my machine though.

Reply:
I'd recommend testing and ensuring that the following works as expected:

- I can associate a label and have it work as the template suggests
- I can use aria-labelledby with a visible label which is not the internal label
- I can successfully set aria-label for a non-visible label

Pulling from: https://w3c.github.io/aria-practices/#radiobutton
> Each radio element is labelled by its content, has a visible label referenced by aria-labelledby, or has a label specified with aria-label.

Reply:
I believe with the latest fixture html all 3 are present and I went through them with VoiceOver